lib/state/weather_request.dart

```diff
@@ -16,3 +17,9 @@ class WeatherRequest with _$WeatherRequest {
   factory WeatherRequest.fromJson(Map<String, dynamic> json) =>
       _$WeatherRequestFromJson(json);
 }
+
+final weatherRequestProvider = StateProvider<WeatherRequest>(
```

Comment:
[nits]
`weatherForecastPanelStateProvider` と命名規則を合わせて、 `weatherRequestStateProvider` としましょう！

```suggestion
final weatherRequestStateProvider = StateProvider<WeatherRequest>(
```